docs/src/components/demo/rehypeDemos.mjs

```diff
@@ -0,0 +1,102 @@
+import { visit } from 'unist-util-visit';
```

Comment:
Add `@ts-check` so it's easy to port this file to TS when it's moved to a separate library.

Reply:
Added (no difference)

Reply:
Actually nah, the types check fail now because a bunch of stuff is untyped.

I am not sure this is a good time to add these because we need to hand-write the types and it seems  a pretty poor experience to author and read them with complex type annotations. Will be easy to add them when we make it into a separate package, do you mind if we keep it as JS right now?

Reply:
All right, let's do it then